packages/xo-web/src/common/xo/migrate-vms-modal/index.js

```diff
@@ -124,10 +120,14 @@ export default class MigrateVmsModalBody extends BaseComponent {
       forEach(vbds, vbd => {
         const vdi = vbd.VDI
         if (!vbd.is_cd_drive && vdi) {
-          mapVdisSrs[vdi] = doNotMigrateVmVdis[vm] || doNotMigrateVdi[vdi] ? this._getObject(vdi).$SR : srId
+          if (!doNotMigrateVmVdis[vm] && !doNotMigrateVdi[vdi]) {
+            mapVdisSrs[vdi] = srId
+          }
         }
       })
-      mapVmsMapVdisSrs[vm] = mapVdisSrs
+      if (!isEmpty(mapVdisSrs)) {
+        mapVmsMapVdisSrs[vm] = mapVdisSrs
```

Comment:
I think the only case where its empty is when we don't want or can't migrate any vdi, no ?

Reply:
It is not empty if at least one VDI needs to be migrated. (if he were on local storage for example)